### PR TITLE
Feat: implement iot device status toggle

### PR DESCRIPTION
**Pull request by rkpattnaik780**
### Type of change

- Enhancement

### Description

IoT Device status toggle implemented in DeviceList and DeviceDetailPage. Filter of device list updated to support boolean values.

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Resolve all warnings
- [ ] Update license
- [ ] Write test cases for null checks in Unit Tests
- [ ] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: console/console-init/ui/src/graphql-module/queries/iot_device.ts [hunk 1]
```diff
@@ -101,16 +101,45 @@ const SORT_RETURN_ALL_DEVICES_FOR_IOT_PROJECT = (sortBy?: ISortBy) => {
   return orderBy;
 };
 
+const concatAND = (filter: string) => {
+  if (filter?.length > 0) return " AND ";
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/graphql-module/queries/iot_device.ts:105]: if condition needs to be modified. since it is a string we should replace the condition with `if(filter?.trim()!==""`

File: console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx [hunk 2]
```diff
@@ -139,10 +150,31 @@ export default function DeviceDetailPage() {
...
+    const dialogType: string = status
+      ? DialogTypes.ENABLE
+      : DialogTypes.DISABLE;
+    dispatch({
+      type: types.SHOW_MODAL,
+      modalType: MODAL_TYPES.UPDATE_DEVICE_STATUS,
+      modalProps: {
+        onConfirm: onConfirmToggleDeviceStatus,
+        selectedItems: [deviceId],
+        option: DialogTypes.DISABLE,
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:172]: It should be dailogType which is defined at line 163

File: console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx [hunk 3]
```diff
@@ -139,10 +150,31 @@ export default function DeviceDetailPage() {
...
+        data: { deviceId, status },
+        detail: getDetailForDialog([{ deviceId }], dialogType),
+        header: getHeaderForDialog([{ deviceId }], dialogType)
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:175]: To be consistent with other screens the confirm button should be in red color. Please add 
`iconType: "danger",
 confirmButtonLabel:"Enable" or "Disable" ` in the modalProps

Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:175]: Do we really need a danger icon to change status of device?

File: console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx [hunk 4]
```diff
@@ -159,46 +152,21 @@ export const DeviceListContainer: React.FC<IDeviceListContainerProps> = ({
...
 
-  const disableDevice = (row: any) => {
-    const { deviceId, jsonData, viaGateway } = row.originalData;
-    const device = {
-      deviceId,
-      jsonData,
-      viaGateway,
-      enabled: false
-    };
+  const toggleDeviceStatus = (row: any, status: boolean) => {
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx:155]: If we are referring to table row here, type of table row could have been used instead of any.

File: console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx [hunk 5]
```diff
@@ -213,7 +184,9 @@ export const DeviceListContainer: React.FC<IDeviceListContainerProps> = ({
       {
         title: enabled ? "Disable" : "Enable",
         onClick: () =>
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx:186]: it can be more simple like 
onClick:()=> toggleDeviceStatus(rowData, enabled?false:true)

File: console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx [hunk 6]
```diff
@@ -163,12 +167,26 @@ export default function DeviceListPage() {
     await setDeleteDeviceQueryVariables(variable);
   };
 
-  const onConfirmEnableSelectedDevices = () => {
-    // TODO: TO BE DONE AFTER BACKEND IS READY
+  const onConfirmEnableSelectedDevices = async (devices: IDevice[]) => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx:170]: don't need to create seperate function for enabled and disabled status.
Create single function and handled both status

const onConfirmStatus=async (devices: IDevice[],status:boolean) => {
  setIsAllSelected(false);
    setSelectedDevices([]);
    const variable = {
      a: { name: projectname, namespace },
      b: devices.map(({ deviceId }) => deviceId),
      status
    };
    await setUpdateDeviceQueryVariables(variable);
}

File: console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx [hunk 7]
```diff
@@ -147,8 +151,8 @@ export default function DeviceListPage() {
     }
   };
 
-  const runFilter = () => {
-    setAppliedFilter(filter);
+  const runFilter = (filter2: IDeviceFilter) => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device/DeviceListPage.tsx:154]: can you please rename param filter2.

File: console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx [hunk 8]
```diff
@@ -229,7 +230,7 @@ const DeviceDetailHeader: React.FunctionComponent<IDeviceDetailHeaderProps> = ({
       <DeviceDetailLayout />
       <SplitItem isFilled />
       <SplitItem className={css(styles.kebab_toggle_margin)}>
-        <SwitchWithToggle
+        <Switch
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:233]: why did you change SwitchWithToggle to Switch ?

Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-device-detail/components/DeviceDetailHeader/DeviceDetailHeader.tsx:233]: SwitchWithToggle does not wait till action from wizard is taken, it changes state on its own.

File: console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx (hunk 5 above)
Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-device/containers/DeviceListContainer/DeviceListContainer.tsx:186]: It had occured to me! But it seems bit weird to pass arguments conditionally as it has not been so anywhere else in the codebase.
